sidecar/app/query/ipa.py

```diff
@@ -248,32 +237,31 @@ def build_from_query(cls, query: IPAQuery):
 
     def run(self):
         settings = get_settings()
-        sidecar_urls = [
-            helper.sidecar_url
-            for helper in settings.helpers.values()
-            if helper.role != Role.COORDINATOR
-        ]
-        for sidecar_url in sidecar_urls:
-            url = urlunparse(
-                sidecar_url._replace(
-                    scheme="https", path=f"/start/{self.query_id}/status"
-                ),
-            )
+        for helper in settings.other_helpers:
+            max_unknonwn_status_wait_time = 100
+            current_unknown_status_wait_time = 0
+            loop_wait_time = 1
             while True:
-                r = httpx.get(url).json()
-                status = r.get("status")
+                status = helper.get_current_query_status(self.query_id)
                 match status:
-                    case Status.IN_PROGRESS.name:
+                    case Status.IN_PROGRESS:
                         break
-                    case Status.KILLED.name:
-                        self.success = False
-                        return
-                    case Status.NOT_FOUND.name:
-                        self.success = False
-                        return
-                    case Status.CRASHED.name:
+                    case Status.KILLED | Status.NOT_FOUND | Status.CRASHED:
                         self.success = False
                         return
+                    case Status.STARTING | Status.COMPILING | Status.WAITING_TO_START:
+                        # keep waiting while it's in a startup state
+                        continue
+                    case Status.UNKNOWN | Status.NOT_FOUND:
+                        # eventually fail if the status is unknown or not found
+                        # for ~100 seconds
+                        current_unknown_status_wait_time += loop_wait_time
+                        if (
+                            current_unknown_status_wait_time
+                            >= max_unknonwn_status_wait_time
+                        ):
+                            self.success = False
```

Comment:
would that inform the UI that we failed to get a "good" status from helper and this is the reason why we couldn't start a query?

Reply:
when a `Step` finishes with `self.success = False`, it causes the `Query` to crash. The UI will show the report collector as crashed, and in the logs we'll see that the `IPACoordinatorWaitForHelpersStep` failed.